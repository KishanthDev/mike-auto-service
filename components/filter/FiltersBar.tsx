"use client";

import { useState } from "react";
import { Filter } from "lucide-react";

const filterOptions = [
  "Open Now",
  "Top Rated",
  "Within 2 miles",
  "Free Estimates",
  "Price: $$$",
];

export const FiltersBar = () => {
  const [activeFilters, setActiveFilters] = useState<string[]>([]);

  const toggleFilter = (filter: string) => {
    setActiveFilters((prev) =>
      prev.includes(filter)
        ? prev.filter((f) => f !== filter)
        : [...prev, filter]
    );
  };

  return (
    <div className="bg-gray-100 dark:bg-gray-900 p-4 rounded-lg mb-6">
      <div className="flex items-center gap-3 flex-wrap">
        <div className="flex items-center gap-2 font-medium text-gray-700 dark:text-gray-200">
          <Filter className="w-4 h-4" />
          Filters:
        </div>
        {filterOptions.map((filter) => (
          <button
            key={filter}
            onClick={() => toggleFilter(filter)}
            className={`px-4 py-2 rounded-full text-sm font-medium transition-colors
              ${
                activeFilters.includes(filter)
                  ? "bg-blue-500 text-white"
                  : "bg-white text-gray-800 hover:bg-gray-200 dark:bg-gray-800 dark:text-gray-200 dark:hover:bg-gray-700"
              }`}
          >
            {filter}
          </button>
        ))}
      </div>
    </div>
  );
};
