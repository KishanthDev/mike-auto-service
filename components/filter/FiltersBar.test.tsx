import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { FiltersBar } from './FiltersBar';

// Mock the Lucide icon
jest.mock('lucide-react', () => ({
  Filter: jest.fn(() => <svg data-testid="filter-icon" />),
}));

describe('FiltersBar Component', () => {
  const filterOptions = [
    'Open Now',
    'Top Rated',
    'Within 2 miles',
    'Free Estimates',
    'Price: $$$',
  ];

  it('renders the filters bar with all options', () => {
    render(<FiltersBar />);
    
    // Check for the header section with icon
    expect(screen.getByText('Filters:')).toBeInTheDocument();
    expect(screen.getByTestId('filter-icon')).toBeInTheDocument();
    
    // Check all filter options are rendered
    filterOptions.forEach(option => {
      expect(screen.getByText(option)).toBeInTheDocument();
    });
  });

  it('applies inactive styles to filters by default', () => {
    render(<FiltersBar />);
    
    filterOptions.forEach(option => {
      const button = screen.getByText(option);
      expect(button).toHaveClass('bg-white');
      expect(button).toHaveClass('text-gray-800');
      expect(button).toHaveClass('dark:bg-gray-800');
      expect(button).toHaveClass('dark:text-gray-200');
    });
  });

  it('toggles filter active state when clicked', () => {
    render(<FiltersBar />);
    const firstFilter = screen.getByText('Open Now');
    
    // Click to activate
    fireEvent.click(firstFilter);
    expect(firstFilter).toHaveClass('bg-blue-500');
    expect(firstFilter).toHaveClass('text-white');
    
    // Click to deactivate
    fireEvent.click(firstFilter);
    expect(firstFilter).toHaveClass('bg-white');
    expect(firstFilter).toHaveClass('text-gray-800');
  });

  it('maintains multiple active filters', () => {
    render(<FiltersBar />);
    const firstFilter = screen.getByText('Open Now');
    const secondFilter = screen.getByText('Top Rated');
    
    // Activate both filters
    fireEvent.click(firstFilter);
    fireEvent.click(secondFilter);
    
    expect(firstFilter).toHaveClass('bg-blue-500');
    expect(secondFilter).toHaveClass('bg-blue-500');
    
    // Check that other filters remain inactive
    const thirdFilter = screen.getByText('Within 2 miles');
    expect(thirdFilter).toHaveClass('bg-white');
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<FiltersBar />);
    expect(asFragment()).toMatchSnapshot();
  });
});