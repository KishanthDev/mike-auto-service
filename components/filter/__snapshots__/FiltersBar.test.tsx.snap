// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiltersBar Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-gray-100 dark:bg-gray-900 p-4 rounded-lg mb-6"
  >
    <div
      class="flex items-center gap-3 flex-wrap"
    >
      <div
        class="flex items-center gap-2 font-medium text-gray-700 dark:text-gray-200"
      >
        <svg
          data-testid="filter-icon"
        />
        Filters:
      </div>
      <button
        class="px-4 py-2 rounded-full text-sm font-medium transition-colors
              bg-white text-gray-800 hover:bg-gray-200 dark:bg-gray-800 dark:text-gray-200 dark:hover:bg-gray-700"
      >
        Open Now
      </button>
      <button
        class="px-4 py-2 rounded-full text-sm font-medium transition-colors
              bg-white text-gray-800 hover:bg-gray-200 dark:bg-gray-800 dark:text-gray-200 dark:hover:bg-gray-700"
      >
        Top Rated
      </button>
      <button
        class="px-4 py-2 rounded-full text-sm font-medium transition-colors
              bg-white text-gray-800 hover:bg-gray-200 dark:bg-gray-800 dark:text-gray-200 dark:hover:bg-gray-700"
      >
        Within 2 miles
      </button>
      <button
        class="px-4 py-2 rounded-full text-sm font-medium transition-colors
              bg-white text-gray-800 hover:bg-gray-200 dark:bg-gray-800 dark:text-gray-200 dark:hover:bg-gray-700"
      >
        Free Estimates
      </button>
      <button
        class="px-4 py-2 rounded-full text-sm font-medium transition-colors
              bg-white text-gray-800 hover:bg-gray-200 dark:bg-gray-800 dark:text-gray-200 dark:hover:bg-gray-700"
      >
        Price: $$$
      </button>
    </div>
  </div>
</DocumentFragment>
`;
